.github/workflows/test.yml

```diff
@@ -97,7 +100,26 @@ jobs:
       - name: cargo test
         uses: actions-rs/cargo@v1
         if: ${{ !matrix.cross }}
+        env:
+          NEXTEST_PROFILE: ci   # defined in .config/nextest.toml
         with:
-          command: test
-          args:  --workspace --target=${{ matrix.target }}
+          command: nextest
+          args:  run --workspace --target=${{ matrix.target }}
           use-cross: ${{ matrix.cross }}
+
+      - name: upload test results
+        uses: actions/upload-artifact@v3
+        if: ${{ !matrix.cross && ( success() || failure() ) }}
```

Comment:
Looks like I misread when `success()` is `true`, I thought it would be false when the workflow was cancelled. Re-reading the docs it is entirely redundant here.